In [10]:
import numpy as np

Вы работаете в отделе маркетинга пищевой компании MyCrunch, которая разрабатывает новый вид вкусных, полезных злаков под названием CrunchieMunchies.

Вы хотите продемонстрировать потребителям, насколько полезны ваши хлопья по сравнению с другими ведущими брендами, поэтому вы собрали данные о питании нескольких разных конкурентов.

Ваша задача - использовать вычисления Numpy для анализа этих данных и доказать, что ваши СrunchieMunchies - самый здоровый выбор для потребителей.

In [11]:
calorie_stats = np.loadtxt("./data/cereal.csv", delimiter=",")
calorie_stats

array([ 70., 120.,  70.,  50., 110., 110., 110., 130.,  90.,  90., 120.,
       110., 120., 110., 110., 110., 100., 110., 110., 110., 100., 110.,
       100., 100., 110., 110., 100., 120., 120., 110., 100., 110., 100.,
       110., 120., 120., 110., 110., 110., 140., 110., 100., 110., 100.,
       150., 150., 160., 100., 120., 140.,  90., 130., 120., 100.,  50.,
        50., 100., 100., 120., 100.,  90., 110., 110.,  80.,  90.,  90.,
       110., 110.,  90., 110., 140., 100., 110., 110., 100., 100., 110.])

In [12]:
average_calories = calorie_stats.mean()
print(f"Mean: {average_calories}")
print(f"The average number of calories among competitors is {average_calories - 60} higher")

Mean: 106.88311688311688
The average number of calories among competitors is 46.883116883116884 higher


In [13]:
calorie_stats_sorted = np.sort(calorie_stats)
print(calorie_stats_sorted)

[ 50.  50.  50.  70.  70.  80.  90.  90.  90.  90.  90.  90.  90. 100.
 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100. 100.
 100. 100. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110. 110.
 110. 110. 110. 120. 120. 120. 120. 120. 120. 120. 120. 120. 120. 130.
 130. 140. 140. 140. 150. 150. 160.]


In [ ]:
higher_avg = calorie_stats[calorie_stats > average_calories]
print(len(calorie_stats))
print(len(higher_avg))

77
47


In [23]:
median_index = len(calorie_stats) // 2 + 1
median_calories = calorie_stats_sorted[median_index]
print("Median:", median_calories)

Median: 110.0


In [29]:
percentiles = np.arange(0, 101, 5)
values = np.percentile(calorie_stats_sorted, percentiles)

nth_percentile = None

for p, val in zip(percentiles, values):
    print(f"{p} percentile: {val} calories")
    if val > 60 and nth_percentile is None:
        nth_percentile = p

0 percentile: 50.0 calories
5 percentile: 70.0 calories
10 percentile: 90.0 calories
15 percentile: 90.0 calories
20 percentile: 100.0 calories
25 percentile: 100.0 calories
30 percentile: 100.0 calories
35 percentile: 100.0 calories
40 percentile: 110.0 calories
45 percentile: 110.0 calories
50 percentile: 110.0 calories
55 percentile: 110.0 calories
60 percentile: 110.0 calories
65 percentile: 110.0 calories
70 percentile: 110.0 calories
75 percentile: 110.0 calories
80 percentile: 120.0 calories
85 percentile: 120.0 calories
90 percentile: 124.00000000000006 calories
95 percentile: 140.0 calories
100 percentile: 160.0 calories


In [31]:
print("The smallest percentile the value of which is greater than 60: ",nth_percentile)

The smallest percentile the value of which is greater than 60:  5


In [33]:
more_calories = len(calorie_stats[calorie_stats > 60]) / len(calorie_stats) * 100
print(f"{more_calories} %")

96.1038961038961 %


In [36]:
calorie_std = calorie_stats_sorted.std()
print("Std: ",calorie_std)

Std:  19.35718533390827


The CrunchieMunchies bar has 60 calories, which is much lower than most other bars. The average calorie content of competitors’ bars is about 107, and the median is 110. Around 96% of the bars have more calories than 60, so the CrunchieMunchies bar is a lighter option. Mycrunch could use this to market the bar as a lower-calorie snack